src/cli/commands/files/add.js

```diff
@@ -94,26 +107,62 @@ module.exports = {
       }
     }
 
-    createAddStream((err, addStream) => {
-      if (err) {
+    // if there are piped arguments, input is data to publish instead of a file
+    // path
+    if(hasPipedArgs) {
+      const data = argv.file
```

Comment:
Yes, the whole file is going to be buffered before the command is triggered. I assumed that since this is for a CLI interface, it would make sense to do so. So it's better to buffer the file using streams and make it async?